### PR TITLE
GanttProject: add mainstream 2.8.9

### PR DESCRIPTION
**Pull request by giansalvo**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.4 17E202
Xcode 9.3.1 9E501 

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by giansalvo** [comment]
I need help: the mainstream developer is using a kind of gradle wrapper that download what needed before building the .app bund.
When I try to build I get the following errors, how should I go on? Thanks for your help.
```
Exception in thread "main" java.lang.RuntimeException: Could not create parent directory for lock file /opt/local/var/macports/home/.gradle/wrapper/dists/gradle-4.7-bin/73xux3q4721jgr4pjvqf9o30/gradle-4.7-bin.zip.lck
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:43)
	at org.gradle.wrapper.Install.createDist(Install.java:48)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:107)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
Command failed: ./gradlew updatelibs distmac
Exit code: 1
Warning: The following existing files were hidden from the build system by trace mode:
  /var/db/timezone/icutz
  /var/db/timezone/zoneinfo/Europe/Rome
The following file would have been hidden from the build system by trace mode if it existed:
  /Network/Library/Java/Extensions/meta-index

```

File: devel/GanttProject/Portfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+github.setup        bardsoftware ganttproject 370b88ac6023dcc51469406063b0b1292c92cdcf
+
+name                GanttProject
+version             2.8.9-git
+
+java.version        1.8
+java.fallback       openjdk8
+
+revision            0
```
Comment (mf2k) [inline devel/GanttProject/Portfile]: Move the `revision` line to just under the `version` line.

File: devel/GanttProject/Portfile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    management tool. It is written in Java/Swing. \
+    GanttProject was started by Alexandre Thomas in 2003 and many \
+    people participated in the development since that. \
+    It is translated in more than 20 languages.
+
+homepage            https://www.ganttproject.biz/
+
+#master_sites        https://github.com/bardsoftware/ganttproject/archive/
+#distname            ${github.project}-${github.version}
+#worksrcdir          ganttproject-ganttproject-${version}
```
Comment (mf2k) [inline devel/GanttProject/Portfile]: Remove these 3 commented out lines.

File: devel/GanttProject/Portfile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+
+checksums           rmd160  f5faf2632cf46dc06cb47032deb249b488c830c5 \
+                    sha256  1792967327b26d7fafc9ce7c3cdd937622b41cd5832bcc4ca2376f6ac9fe248f \
+                    size    15429698
+
+depends_lib         port:gradle
```
Comment (mf2k) [inline devel/GanttProject/Portfile]: Switch this to `depends_lib-append` to not overwrite portgroup dependencies.

File: office/GanttProject/Portfile [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           java 1.0
+
+github.setup        bardsoftware ganttproject 370b88ac6023dcc51469406063b0b1292c92cdcf
+
+name                GanttProject
+version             2.8.9-git
```
Comment (cjones051073) [inline office/GanttProject/Portfile]: Please remove the -git part about of the version.

File: office/GanttProject/Portfile [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+checksums           rmd160  f5faf2632cf46dc06cb47032deb249b488c830c5 \
+                    sha256  1792967327b26d7fafc9ce7c3cdd937622b41cd5832bcc4ca2376f6ac9fe248f \
+                    size    15429698
+
+depends_lib-append  port:gradle
+
+use_configure       no
+
+build {
+    set ::env(JAVA_HOME) [exec /usr/libexec/java_home -v ${java.version}]
```
Comment (cjones051073) [inline office/GanttProject/Portfile]: The JAVA_HOME variable is set by the java portgroup, so this line should not be required.

**Comment by cjones051073** [comment]
Agreed.

**Pull request closed without merging**